src/tests/blend_tests.cpp

```diff
@@ -22,20 +22,58 @@ static constexpr uint32_t kDefaultDMAColorChannel = 9;
 static constexpr uint32_t kCheckerboardA = 0xFF202020;
 static constexpr uint32_t kCheckerboardB = 0xFF000000;
 
-static constexpr const char kSignedAddName[] = "SignedAdd";
-static constexpr const char kSignedRevSubtractName[] = "SignedRevSubtract";
+struct BlendFunc {
+  const char *name;
+  uint32_t value;
+};
+
+static constexpr struct BlendFunc kBlendEqns[] = {
+    {"ADD", NV097_SET_BLEND_EQUATION_V_FUNC_ADD},
+    {"SUB", NV097_SET_BLEND_EQUATION_V_FUNC_SUBTRACT},
+    {"REVSUB", NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT},
+    {"MIN", NV097_SET_BLEND_EQUATION_V_MIN},
+    {"MAX", NV097_SET_BLEND_EQUATION_V_MAX},
+    {"SADD", NV097_SET_BLEND_EQUATION_V_FUNC_ADD_SIGNED},
+    {"SREVSUB", NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT_SIGNED},
+};
+
+static constexpr struct BlendFunc kBlendFactors[] = {
+    {"0", NV097_SET_BLEND_FUNC_SFACTOR_V_ZERO},
+    {"1", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE},
+    {"srcRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_COLOR},
+    {"1-srcRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_SRC_COLOR},
+    {"srcA", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA},
+    {"1-srcA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_SRC_ALPHA},
+    {"dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_ALPHA},
+    {"1-dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_ALPHA},
+    {"dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_COLOR},
+    {"1-dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_COLOR},
+    {"srcAsat", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA_SATURATE},
+    {"ctRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_COLOR},
+    {"1-ctRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_COLOR},
+    {"ctA", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_ALPHA},
+    {"1-ctA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_ALPHA},
+};
 
 static constexpr uint32_t kTextureSize = 256;
 static constexpr uint32_t kTexturePitch = kTextureSize * 4;
 
 BlendTests::BlendTests(TestHost &host, std::string output_dir, const Config &config)
     : TestSuite(host, std::move(output_dir), "Blend tests", config) {
-  tests_[kSignedAddName] = [this]() {
-    Test(kSignedAddName, [this]() { Body(NV097_SET_BLEND_EQUATION_V_FUNC_ADD_SIGNED); });
-  };
-  tests_[kSignedRevSubtractName] = [this]() {
-    Test(kSignedRevSubtractName, [this]() { Body(NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT_SIGNED); });
-  };
+  for (const auto &test_eqn : kBlendEqns) {
+    std::string name = "";
+    uint32_t factor = kBlendFactors[1].value;
+    name += kBlendFactors[1].name;
+    name += "_";
+    name += kBlendFactors[1].name;
+    name += "_";
+    name += test_eqn.name;
+    tests_[name] = [this, name, &test_eqn, factor]() {
+      Test(test_eqn.name, [this, name, &test_eqn, factor]() { 
```

Comment:
Even though they're the same now, let's change everything to pass source and destination factors separately.